src/codewhisperer/models/model.ts

```diff
@@ -81,6 +85,30 @@ export interface InlineCompletionItem {
     index: number
 }
 
+export class Recommendation implements CodeWhispererUserClient.Completion {
+    readonly content: string
+    readonly references?: CodeWhispererUserClient.References | CodeWhispererClient.References | undefined
+    readonly mostRelevantMissingImports?: CodeWhispererClient.Imports | CodeWhispererUserClient.Imports | undefined
+    readonly completionType: CodewhispererCompletionType
+
+    constructor(
+        readonly cwRecommendation: CodeWhispererClient.Recommendation | CodeWhispererUserClient.Completion,
+        public suggestionState: CodewhispererSuggestionState | 'Showed' | undefined = undefined
+    ) {
+        this.content = cwRecommendation.content
+        this.references = cwRecommendation.references
+        this.mostRelevantMissingImports = cwRecommendation.mostRelevantMissingImports
+
+        const nonBlankLines = cwRecommendation.content.split('\n').filter(line => line.trim() !== '').length
```

Comment:
when cwRecommendation.content = `const a = "\n" \n`, this .split('\n') will apply to the "\n"

Reply:
@leigaol  I didn't change the logic which we use to determines the completion type. It's cp and paste from 
https://github.com/aws/aws-toolkit-vscode/pull/4032/files#diff-c5042bc2a059c2ecf6c5dee425ba140b81e6470b60a8907755745bf44c3faaa1L65-L67

If the logic is wrong, let's cut a sperate tt to track it. I want this PR only focusing on structural change.